client/app/dedicatedCloud/datacenter/drp/dedicatedCloud-datacenter-drp.service.js

```diff
@@ -0,0 +1,131 @@
+export default class {
+  /* @ngInject */
+  constructor($q, OvhApiDedicatedCloud, DEDICATEDCLOUD_DATACENTER_DRP_OPTIONS) {
+    this.$q = $q;
+    this.OvhApiDedicatedCloud = OvhApiDedicatedCloud;
+    this.DEDICATEDCLOUD_DATACENTER_DRP_OPTIONS = DEDICATEDCLOUD_DATACENTER_DRP_OPTIONS;
+  }
+
+  getPccIpAddresses(serviceName) {
+    return this.OvhApiDedicatedCloud.Ip().v6().query({
+      serviceName,
+    }).$promise
+      .then(ipAddresses => ipAddresses.map(ipAddress => this.OvhApiDedicatedCloud.Ip().v6()
+        .get({
+          serviceName,
+          network: ipAddress,
+        }).$promise));
+  }
+
+  getPccDrpPlan(serviceName) {
+    return this.OvhApiDedicatedCloud.Datacenter().v6().query({
+      serviceName,
+    }).$promise
+      .then(datacenters => this.$q.all(datacenters
+        .map(datacenterId => this.getDrpState({
+          serviceName,
+          datacenterId,
+        }))));
+  }
+
+  getPccIpAddressesDetails(serviceName) {
+    return this.OvhApiDedicatedCloud.Ip().v6().query({
+      serviceName,
+    }).$promise
+      .then(ipAddresses => this.$q.all(ipAddresses
+        .map(ipAddress => this.OvhApiDedicatedCloud.Ip().Details()
+          .v6()
+          .get({
+            serviceName,
+            network: ipAddress,
+          }).$promise))
+        .then(ipAddressesDetails => _.flatten(ipAddressesDetails)));
+  }
+
+  getDrpState(serviceInformations) {
+    return this.OvhApiDedicatedCloud
+      .Datacenter().Zerto().v6().state(serviceInformations, null).$promise
+      .then(state => angular.extend(state.data, serviceInformations));
```

Comment:
```suggestion
      .then(state => ({...state.data, ...serviceInformations}));
```
Or Lodash, we will get rid of Angular one day, but not Lodash or Javascript :D